gallery/how_to/work_with_microtvm/micro_custom_ide.py

```diff
@@ -0,0 +1,358 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+.. _tutorial-micro-ide:
+
+9. Bring microTVM to your own development environment
+======================================================
+**Author**:
+`Mohamad Katanbaf <https://github.com/mkatanbaf>`_
+
+This tutorial describes the steps required to integrate a model compiled with microTVM into a custom development environment.
+We use `STM32CubeIDE <https://www.st.com/en/development-tools/stm32cubeide.html>`_, as the target IDE in this tutorial, but we do not rely on any specific feature of this IDE and integrating microTVM in other IDEs would be similar.
+We also use the Visual Wake Word (VWW) model from MLPerf Tiny and the nucleo_l4r5zi board here, but the same steps can be used for any other model or target MCU.
+If you want to use another target MCU with the vww model, we recommend a cortex-M4 or cortex-M7 device with ~512 KB and ~256 KB of Flash and RAM respectively.
+
+Here is a brief overview of the steps that we would take in this tutorial.
+
+1. We start by importing the model, compiling it using TVM and generating the `Model Library Format <https://tvm.apache.org/docs/arch/model_library_format.html>`_ (MLF) tar-file that includes the generated code for the model as well as all the required TVM dependencies.
+2. We also add two sample images in binary format (one person and one not-person sample) to the .tar file for evaluating the model.
+3. Next we use the stmCubeMX to generate the initialization code for the project in stmCube IDE.
+4. After that, we include our MLF file and the required CMSIS libraries in the project and build it.
+5. Finally, we Flash the device and evaluate the model performance on our sample images.
+
+Let's Begin.
+"""
+
+######################################################################
+# Install microTVM Python dependencies
+# ------------------------------------
+#
+# TVM does not include a package for Python serial communication, so
+# we must install one before using microTVM. We will also need TFLite
+# to load models, and Pillow to prepare the sample images.
+#
+#   .. code-block:: bash
+#
+#     %%shell
+#     pip install pyserial==3.5 tflite==2.1 Pillow==9.0 typing_extensions
+#
+
+
+######################################################################
+# Import Python dependencies
+# ---------------------------
+#
+import os
+import numpy as np
+import pathlib
+import json
+from PIL import Image
+import tarfile
+
+import tvm
```

Comment:
Do you want to link to or note above that you also need TVM python package installed?

Reply:
also, I guess that needs to be TVM built with USE_MICRO enabled

Reply:
@mehrdadh right, but what if someone want to run locally?

Reply:
I added a link to tvm documentation page for installing tvm.